### PR TITLE
[pickers] Add reference links to misc picker components

### PR DESCRIPTION
**Pull request by michelengelen**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

---

This adds Demo and API links to the component annotation for ...
- `DateTimePickerTabs`
- `DateRangePickerDay`
- `DayCalendarSkeleton`
- `LocalizationProvider`
- `PickersCalendarHeader`
- `PickersDay`
- `PickersActionBar`
- `PickersLayout`
- `PickersShortcuts`
... components

It's part of a series to solve #9226

### CODE REVIEW

File: packages/x-date-pickers/src/PickersCalendarHeader/PickersCalendarHeader.tsx [hunk 1]
```diff
@@ -189,6 +189,15 @@ type PickersCalendarHeaderComponent = (<TDate>(
   props: PickersCalendarHeaderProps<TDate> & React.RefAttributes<HTMLButtonElement>,
 ) => React.JSX.Element) & { propTypes?: any };
 
+/**
+ * Demos:
+ *
+ * - [Custom slots and subcomponents](https://mui.com/x/react-date-pickers/custom-components/)
```
Comment (LukasTy) [inline packages/x-date-pickers/src/PickersCalendarHeader/PickersCalendarHeader.tsx:195]: WDYT about also adding the `DateCalendar` and `DateRangeCalendar` pages as they showcase this component?
```suggestion
 * - [DateCalendar](https://mui.com/x/react-date-pickers/date-calendar/)
 * - [DateRangeCalendar](https://mui.com/x/react-date-pickers/date-range-calendar/)
 * - [Custom slots and subcomponents](https://mui.com/x/react-date-pickers/custom-components/)
```

File: packages/x-date-pickers/src/PickersDay/PickersDay.tsx [hunk 2]
```diff
@@ -495,11 +495,10 @@ PickersDayRaw.propTypes = {
 } as any;
 
 /**
- *
  * Demos:
  *
- * - [Date Picker](https://mui.com/x/react-date-pickers/date-picker/)
- *
+ * - [DatePicker](https://mui.com/x/react-date-pickers/date-picker/)
```
Comment (LukasTy) [inline packages/x-date-pickers/src/PickersDay/PickersDay.tsx:500]: WDYT about removing the `DatePicker` page as it technically doesn't show anything more than the specific `DateCalendar` page? 🤔 
```suggestion
```

File: packages/x-date-pickers/src/PickersShortcuts/PickersShortcuts.tsx [hunk 3]
```diff
@@ -45,6 +45,16 @@ export interface PickersShortcutsProps<TValue> extends ExportedPickersShortcutPr
   isValid: (value: TValue) => boolean;
 }
 
+/**
+ * Demos:
+ *
+ * - [Custom slots and subcomponents](https://mui.com/x/react-date-pickers/custom-components/)
```
Comment (LukasTy) [inline packages/x-date-pickers/src/PickersShortcuts/PickersShortcuts.tsx:51]: WDYT about removing this page?
It does not list any demo of this component but links to the `Shortcuts` page. 🤔 
```suggestion
```

**Review by LukasTy — APPROVED** [review]

**Comment by michelengelen** [comment]
@LukasTy great finds. I did apply your suggestions! 🙇🏼 💯

**Pull request merged**